### PR TITLE
docs: update PR template to include QA labels.

### PR DESCRIPTION
**Pull request by cortisiko**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

This PR aims to update the template, incorporating QA labels, to inform engineers about the usage and criteria for applying QA labels.

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Review by Cal-L — DISMISSED** [review]
Lgtm

**Review by NicolasMassart — DISMISSED** [review]
I like this!

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -9,7 +9,9 @@ Please ensure that any applicable requirements below are satisfied before submit
 2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
 3. Add `needs-dev-review` label when work is completed
 4. Add `needs-qa` label when dev review is completed
-5. Add `QA Passed` label when QA has signed off
+5. Add `No QA/E2E only` when dev review is completed and your PR does not require any manual QA effort
```
Comment (cryptotavares) [inline .github/pull_request_template.md:12]: For `No QA/E2E only` should we say that a link to a successfully e2e test execution in Bitrise is required? 🤔

Comment (cortisiko) [reply .github/pull_request_template.md:12]: Good shout. I will make reference to having passing E2E tests on bitrise.

**Review by cryptotavares — APPROVED** [review]

**Pull request merged**